### PR TITLE
Permission fix

### PR DESCRIPTION
**Pull request by alexfernandez803**
Addressed issue https://github.com/remotion-dev/remotion/issues/1699

### CODE REVIEW

**Comment by alexfernandez803** [comment]
@JonnyBurger I've tried to add an example lambda application into the project but socket security is having issues with `aws-cdk-lib` so I've hosted  it in other repo for now. Let me know your thoughts,

**Comment by JonnyBurger** [comment]
@alexfernandez803 Thanks a lot!
Will now review it, putting the example in another repo is a good idea! For the future, Socket complaints are just warnings, we can dismiss them if necessary.

File: packages/docs/docs/lambda/lambdarolepermissions.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+image: /generated/articles-docs-lambda-lambda-role-permissions.png
+id: lambdarolepermissions
+title: Lambda permissions
```
Comment (JonnyBurger) [inline packages/docs/docs/lambda/lambdarolepermissions.md]: There is already a page called Lambda permissions. Also, I think this tutorial should only be followed if you cannot follow the normal tutorial. Maybe "Using Lambda without IAM" is a better title?

File: packages/docs/docs/lambda/lambdarolepermissions.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+---
+image: /generated/articles-docs-lambda-lambda-role-permissions.png
```
Comment (JonnyBurger) [inline packages/docs/docs/lambda/lambdarolepermissions.md]: Need to generate a new image - delete this line and then run `node extract-articles.mjs` in packages/docs

File: packages/docs/docs/lambda/lambdarolepermissions.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+There also some that has requirements to execute the [`renderMediaOnLambda()`](/docs/lambda/rendermediaonlambda) on services such as [`Lambda`](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html), [`EC2`](https://aws.amazon.com/ec2/) and other [`compute services`](https://aws.amazon.com/products/compute/) which using the long term credentials is not an option. AWS has the concept of [IAM Roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles), when a role is assigned to a AWS service it assumes any elevated privileges from the attached AWS `policies`, it is empowered to execute activities such as putting a file to S3 bucket. The role is given short lived  AWS credentials such as `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_SESSION_TOKEN`, to generate the video. With this approach security is enhanced as we don't have any long term credential lurking around and we don't need to keep track of it's rotation.
+
+The steps below provides authorization to the lambda function to execute `renderMediaOnLambda()` without permission issues.
+
+#### Prerequisites
```
Comment (JonnyBurger) [inline packages/docs/docs/lambda/lambdarolepermissions.md]: Can be a `<h2>` like the other titles

File: packages/docs/docs/lambda/lambdarolepermissions.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+1. This assumes that you have deployed a `lambda` function running in AWS. An example is created in [`here`](https://github.com/remotion-dev/remotion/tree/main/packages/example-lambda) using [`CDK`](https://docs.aws.amazon.com/cdk/v2/guide/work-with-cdk-typescript.html) to give you an idea how to call [`renderMediaOnLambda()`](/docs/lambda/rendermediaonlambda) inside another a lambda function, the function is triggered by [API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/welcome.html). This assumes that you have knowledge on using [`CDK`](https://docs.aws.amazon.com/cdk/v2/guide/work-with-cdk-typescript.html), a [write up](/docs/lambda/serverless-lambdas-setup) for this is also created for anyone to follow.
```
Comment (JonnyBurger) [inline packages/docs/docs/lambda/lambdarolepermissions.md]: Need to link the correct repo here

File: packages/docs/docs/lambda/lambdarolepermissions.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+title: Lambda permissions
+slug: /lambda/lambda-role-permissions
+crumb: "Lambda"
+---
+
+import {UserPolicy} from '../../components/lambda/user-permissions.tsx';
+
+## Context
+
+As documented on [`permission`](/docs/lambda/permissions) we create a remotion user and assign `policies`, this policies gives permissions to the `lambda` to render video from [`renderMediaOnLambda()`](/docs/lambda/rendermediaonlambda). This approach is sometimes not acceptable as AWS client id and secret associated with the user are considered long term credentials and prone for inadvertent leakage.
```
Comment (JonnyBurger) [inline packages/docs/docs/lambda/lambdarolepermissions.md]: We'll go over and improve some grammatical things in these sections - @MehmetAdemi can you help with improving the language in this document?

File: packages/docs/docs/lambda/serverlesslambdasetup.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
+---
+image: /generated/articles-docs-lambda-serverlesslambdasetup.png
```
Comment (JonnyBurger) [inline packages/docs/docs/lambda/serverlesslambdasetup.md]: - Title seems to generic for me - seems to suggest anybody should follow this setup guide, but I see it as only for people who cannot or don't want to use IAM. How about "Example setup without IAM"?
- Let's also regenerate the card like explained in the other comment

File: packages/docs/docs/lambda/serverlesslambdasetup.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+id: serverlesslambdasetup
+title: Serverless lambda setup
+slug: /lambda/serverless-lambdas-setup
+crumb: "Lambda"
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+This is a follow up write up for a serverless function from [lamdbda permissions](/docs/lambda/lambda-role-permissions).
```
Comment (JonnyBurger) [inline packages/docs/docs/lambda/serverlesslambdasetup.md]:
```suggestion
This is a follow up write up for a serverless function from [Lambda permissions](/docs/lambda/lambda-role-permissions).
```

File: packages/docs/docs/lambda/serverlesslambdasetup.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+
+### Prequisites
+- Make sure that you're local AWS profile is able to deploy to AWS.
+
+### Setup
+
+#### 1. Clone or download the project 
+Project is located inside the [`remotion project`](https://github.com/alexfernandez803/example-lambda)
```
Comment (JonnyBurger) [inline packages/docs/docs/lambda/serverlesslambdasetup.md]: Let's call it "example project" or "reference project"

**Review by JonnyBurger — CHANGES_REQUESTED** [review]
- The PNPM lockfile should be reverted to the original since the example project has been removed

File: packages/docs/docs/lambda/lambdarolepermissions.md (hunk 1 above)
Comment (alexfernandez803) [reply packages/docs/docs/lambda/lambdarolepermissions.md]: changed to 'Using Lambda without IAM user' to be more specific.

File: packages/docs/docs/lambda/lambdarolepermissions.md (hunk 3 above)
Comment (alexfernandez803) [reply packages/docs/docs/lambda/lambdarolepermissions.md]: updated the document as per suggestion.

File: packages/docs/docs/lambda/lambdarolepermissions.md (hunk 4 above)
Comment (alexfernandez803) [reply packages/docs/docs/lambda/lambdarolepermissions.md]: made the change

File: packages/docs/docs/lambda/serverlesslambdasetup.md (hunk 8 above)
Comment (alexfernandez803) [reply packages/docs/docs/lambda/serverlesslambdasetup.md]: completed.

File: packages/docs/docs/lambda/lambdarolepermissions.md (hunk 2 above)
Comment (alexfernandez803) [reply packages/docs/docs/lambda/lambdarolepermissions.md]: thanks for this, I did not know that there is a utility for that.

File: packages/docs/docs/lambda/serverlesslambdasetup.md (hunk 6 above)
Comment (alexfernandez803) [reply packages/docs/docs/lambda/serverlesslambdasetup.md]: I've resolved it by updating it to 'Example setup without IAM user'

File: packages/docs/docs/lambda/serverlesslambdasetup.md (hunk 7 above)
Comment (alexfernandez803) [reply packages/docs/docs/lambda/serverlesslambdasetup.md]: I've changed it to 
This is a follow up write up for the serverless function from [using lambda without IAM user](/docs/lambda/using-lambda-without-iam-user).

**Comment by JonnyBurger** [comment]
Thanks for the fixes! We'll do some copywriting in the next days and hope to merge it next week.
Also, noticed some links are broken in Vercel Preview:


```
Exhaustive list of all broken links found:
--
15:56:43.391 |  
15:56:43.391 | - On source page path = /docs/lambda/example-setup-without-iam-user:
15:56:43.392 | -> linking to /docs/lambda/usinglambdawithoutiamuser
15:56:43.392 | -> linking to /docs/lambda/examplesetupwithoutiamuser#output (resolved as: /docs/lambda/examplesetupwithoutiamuser)
15:56:43.392 |  
15:56:43.392 | - On source page path = /docs/lambda/using-lambda-without-iam-user:
15:56:43.392 | -> linking to /docs/lambda/examplesetupwithoutiamuser
15:56:43.392 | -> linking to /docs/lambda/trigger-lambda-from-aws#setup (resolved as: /docs/lambda/trigger-lambda-from-aws)
```

**Comment by MehmetAdemi** [comment]
Hi @alexfernandez803, can I start with the proofreading?

**Comment by alexfernandez803** [comment]
> Hi @alexfernandez803, can I start with the proofreading?

yes , please go ahead.

**Comment by MehmetAdemi** [comment]
I had a read on two docs. @alexfernandez803 could you check if my input and changes make sense?

**Comment by alexfernandez803** [comment]
Thanks @MehmetAdemi, it's much better now than my version. I have some wording fixes and added a section for the remotion dependency on the example project.

**Comment by MehmetAdemi** [comment]
Awesome, thanks @alexfernandez803!

**Comment by JonnyBurger** [comment]
Thanks @alexfernandez803 for the great article and @MehmetAdemi for the copywriting!
Ready to ship and bounty can be paid out 🚀

**Review by JonnyBurger — APPROVED** [review]

**Comment by MehmetAdemi** [comment]
It's fun to get things done like this!

**Pull request merged**